packages/sdk-middleware-http/src/http.js

```diff
@@ -64,28 +63,46 @@ export default function createHttpMiddleware({
     retryDelay = 200,
     maxDelay = Infinity,
   } = {},
+  fetch: fetcher,
 }: HttpMiddlewareOptions): Middleware {
+  if (!fetcher && typeof fetch === 'undefined')
+    throw new Error(
+      '`fetch` is not available. Please pass in `fetch` as an option or have it globally available.'
+    )
+
+  if (!fetcher)
+    // `fetcher` is set here rather than the destructuring to ensure fetch is
+    // declared before referencing it otherwise it would cause a `ReferenceError`.
+    // For reference of this pattern: https://github.com/apollographql/apollo-link/blob/498b413a5b5199b0758ce898b3bb55451f57a2fa/packages/apollo-link-http/src/httpLink.ts#L49
+
+    // eslint-disable-next-line
+    fetcher = fetch
+
   return next => (request: MiddlewareRequest, response: MiddlewareResponse) => {
     const url = host.replace(/\/$/, '') + request.uri
     const body =
       typeof request.body === 'string' || Buffer.isBuffer(request.body)
         ? request.body
-        : JSON.stringify(request.body)
+        : // NOTE: `stringify` of `null` gives the String('null')
+          JSON.stringify(request.body || undefined)
     const requestHeader = {
-      'Content-Type': 'application/json',
+      'Content-Type': ['application/json'],
```

Comment:
How I read the whatwg-fetch-spec passing an array actually does not seem to work. However it works: 

`await fetch(location.href, { headers: { 'Content-Type': ['application/json', 'application/javascript']}})`

leads to a concatenated string for `Content-Type` so 🤷‍♂️

Reply:
Good to know. I guess the tests are focussed on `node-fetch` (which is what's passed under test).

Reply:
Hmm I'm confused on why we need this 🧐

Reply:
How a list of headers for one key is mangled differs among the `fetch` implementations. `node-fetch` makes them an array by default. As a result we align our expectations with it. Does that make sense? If we'd use a native `fetch` in the browser a list of headers for a given key would lead to a concatenated string.

Reply:
Ok but just to be clear: it works also with just a string right?